zerver/tests/test_openapi.py

```diff
@@ -276,6 +287,68 @@ def check_for_non_existant_openapi_endpoints(self) -> None:
                 msg += "\n + {}".format(undocumented_path)
             raise AssertionError(msg)
 
+    def _highest_priority_type(self, types: List[type]) -> type:
+        priority = {list: 1, str: 2, int: 3, bool: 4}
+        tyiroirp = {1: list, 2: str, 3: int, 4: bool}
+        val = 5
+        for t in types:
+            v = priority.get(t, 5)
+            if v < val:
+                val = v
+        return tyiroirp.get(val, types[0])
+
+    def _get_standardized_argument_type(self, t: Any) -> type:
+        origin = getattr(t, "__origin__", None)
+        if not origin:
+            # Then it's most likely one of the fundamental data types
+            # I.E. Not one of the data types from the "typing" module.
+            return t
+        elif origin == Union:
+            subtypes = []
+            for st in t.__args__:
+                subtypes.append(self._get_standardized_argument_type(st))
+            return self._highest_priority_type(subtypes)
+        elif origin == List:
+            return list
+        return self._get_standardized_argument_type(t.__args__[0])
+
+    def check_argument_types(self, function: Callable[..., HttpResponse],
+                             openapi_parameters: List[Dict[str, Any]]) -> None:
+        op = set([(element["name"], VARMAP[element["schema"]["type"]]) for
+                  element in openapi_parameters])
+
+        fp = set()
+        while getattr(function, "__wrapped__", None):
+            function = function.__wrapped__  # type: ignore # mypy doesn't recognize the condition
+        for vname, defval in inspect.signature(function).parameters.items():
+            defval = defval.default
+            if defval.__class__ == REQ:
+                vtype = self._get_standardized_argument_type(function.__annotations__[vname])
+                vname = defval.post_var_name  # type: ignore # it's a REQ; account for whence
```

Comment:
If you do a `type: REQ` on the previous line, I think you won't need this ignore.

Reply:
It doesn't really work...
>mypy      | zerver/tests/test_openapi.py:363: error: Invalid type "zerver.lib.request.REQ"
mypy      | zerver/tests/test_openapi.py:364: error: REQ? has no attribute "__class__"
mypy      | zerver/tests/test_openapi.py:366: error: REQ? has no attribute "post_var_name"

Reply:
For fairly deep mypy internal reasons (basically making type-checking inside our view functions work nicely), `REQ` has a technically incorrect type stub, defined in `zerver/lib/request.pyi`.  So it's possible my previous statement was wrong and there's no way to do this properly without ignores.